### PR TITLE
MC-30104: When using MysqlMQ messages are always set to complete even if exception occurs

### LINKED ISSUE #18140: When using MysqlMQ messages are always set to complete even if exception occurs
### Preconditions (*)

1. Magento 2.3 and 2.4-develop
2. Using mysql mq driver to work with the queue

### Steps to reproduce (*)

1. Publish any message to the queue
2. In the consumer throw Exception

### Expected result (*)

1.Message should be flagged as error in the queue

### Actual result (*)

1. Message is set to complete and ACK

Problem is here -

When exception happens in the Consumer message is flagged as error here 
https://github.com/magento/magento2/blob/2.4-develop/lib/internal/Magento/Framework/MessageQueue/Consumer.php#L229

but then this status is overwritten by

 https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/MysqlMq/Model/Driver/Queue.php#L114

### PR DESCRIPTION
**Pull request by engcom-Golf**
### Description (*)

Stores lock id in lock object after creation
### Fixed Issues (if relevant)

1. Fixes magento/magento2#18140

### Manual testing scenarios (*)

1. Add to \Magento\Catalog\Model\Attribute\Backend\Consumer::process exception throw
2. Create product
3. Update attribute by massaction
4. perform cron:run 3 times command in case if cron is not installed
#### Expected result:
Status of message in queue_message_status table = 6 (error)

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Golf** [comment]
@magento run all tests

File: dev/tests/integration/testsuite/Magento/MessageQueue/Model/ResourceModel/LockTest.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+use Magento\Framework\MessageQueue\LockInterface;
+use Magento\TestFramework\ObjectManager;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Covers Lock resource model test cases
+ */
+class LockTest extends TestCase
+{
+    public function testSaveLock()
```
Comment (engcom-Kilo) [inline dev/tests/integration/testsuite/Magento/MessageQueue/Model/ResourceModel/LockTest.php:19]: This test doesn't cover initial test case. Please add more specific test.

**Review by engcom-Kilo — CHANGES_REQUESTED** [review]

**Comment by engcom-Golf** [comment]
@magento run all tests

File: dev/tests/integration/testsuite/Magento/MessageQueue/Model/ResourceModel/LockTest.php (hunk 1 above)
Comment (engcom-Golf) [reply dev/tests/integration/testsuite/Magento/MessageQueue/Model/ResourceModel/LockTest.php:19]: added test that covers original case

File: dev/tests/integration/testsuite/Magento/MessageQueue/Model/ConsumerTest.php [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    {
+        $this->objectManager = ObjectManager::getInstance();
+        /** @var ConsumerFactory $factory */
+        $factory = $this->objectManager->get(ConsumerFactory::class);
+        $this->model = $factory->get('demoConsumerWithConnectionException');
+        $this->queueResource = $this->objectManager->get(Queue::class);
+    }
+
+    /**
+     * Tests running of specific consumer check status if after retry is not in success status but left in progress
```
Comment (engcom-Kilo) [inline dev/tests/integration/testsuite/Magento/MessageQueue/Model/ConsumerTest.php]: This description is confusing. Please add more clear one.

**Review by engcom-Kilo — APPROVED** [review]

File: dev/tests/integration/testsuite/Magento/MessageQueue/Model/ConsumerTest.php (hunk 2 above)
Comment (engcom-Golf) [reply dev/tests/integration/testsuite/Magento/MessageQueue/Model/ConsumerTest.php]: updated

**Comment by engcom-Golf** [comment]
@magento run all tests

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-31480`.

**Pull request merged**